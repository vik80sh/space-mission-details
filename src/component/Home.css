.home-wrapper {
    padding-top: 50px;
    margin: 0;
}

.home-wrapper .filter-section {
    background-color: #ffffff;
    padding: 10px 10px 20px;
}

.home-wrapper .filter-section .button-selected {
    background-color: green;
}

.home-wrapper .filter-section .filter-title {
    border-bottom: 1px solid gray;
    width: 100%;
    margin: 0 30px;
    font-size: 16px;
}

.home-wrapper .filter-section .successfull-launch {
    padding-top: 25px !important;
}
.mission-details{
    margin-bottom: -10px;
}
.mission-details .card-wrapper {
    padding: 0;
}

.mission-details .card-wrapper .card-container {
    float: left;
    margin: 0 5px 10px;
    padding: 15px;
    background-color: #ffffff;
}

.mission-details .card-wrapper .card-container p {
    padding: 0;
    margin: 0;
    padding-bottom: 5px;
}

.card-wrapper .card-container .image-wrapper {
    background-color: silver;
}

.card-wrapper .card-container .image-wrapper img {
    width: 100%;
    height: 100%;
}

.mission-name {
    padding-top: 10px !important;
    color: cornflowerblue;
}

.mission-id {
    padding-bottom: 0 !important;
}

.mission-id-res {
    margin: 0 5px 0 15px;
    padding-bottom: 10px;
    color: blueviolet;
}

p span {
    color: blueviolet;
}

.spinner-loading{
    margin-top: 50px;
    width: 100%;
    position: absolute;
    z-index: 1;
    top: 50%;
    left: 0;
    bottom: 0;
  }

  .loader {
    border: 16px solid #f3f3f3;
    border-radius: 50%;
    border-top: 16px solid #aedb34;
    width: 100px;
    height: 100px;
    -webkit-animation: spin 2s linear infinite;
    /* Safari */
    animation: spin 2s linear infinite;
    margin: 0 auto;
  }

  /* Safari */
  @-webkit-keyframes spin {
    0% {
      -webkit-transform: rotate(0deg);
    }

    100% {
      -webkit-transform: rotate(360deg);
    }
  }

  @keyframes spin {
    0% {
      transform: rotate(0deg);
    }

    100% {
      transform: rotate(360deg);
    }
  }
@media (max-width:  575px) {
    .home-wrapper .filter-section {
        margin-bottom: 10px;
    }

    .home-wrapper .filter-section .filter-title {
        font-size: 24px;
    }

    .mission-details {
        padding: 0 !important;
        margin: 0 !important;
    }

    .mission-details .card-wrapper .card-container {
        margin: 10px 0;
    }
}