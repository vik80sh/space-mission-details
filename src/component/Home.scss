@import './../BasicStyle/mixing.scss';

.home-wrapper {
    padding-top: 50px;
    margin: 0;

    .filter-section {
        background-color: #ffffff;
        padding: 10px 10px 20px;

        .text-center {
            padding: 0 3px;
        }

        .button-selected {
            background-color: green;
        }

        .filter-title {
            border-bottom: 1px solid gray;
            width: 100%;
            margin: 0 30px;
            font-size: 16px;
        }

        .successfull-launch {
            padding-top: 25px !important;
        }
    }

    .mission-details {
        margin-bottom: -10px;

        .card-wrapper {
            padding: 10px 0 0 0 !important;

            @include tablet {
                padding: 0 !important;
                display: flex;
            }

            @include desktop {
                padding: 0 !important;
            }

            .card-data {
                @include tablet {
                    flex-basis: 50%;
                }

                @include desktop {
                    flex-basis: 25%;
                }

                .card-conatainer {
                    float: left;
                    margin: 0 0px 10px;
                    padding: 15px;
                    background-color: #ffffff;

                    @include tablet {
                        margin: 0 5px 10px;
                    }

                    @include desktop {
                        margin: 0 5px 10px;

                    }

                    p {
                        padding: 0;
                        margin: 0;
                        padding-bottom: 5px;
                    }

                    .image-wrapper {
                        background-color: silver;
                        height: 320px;

                        @include tablet {
                            height: 330px;
                        }

                        @include desktop {
                            height: 216px;
                        }

                        img {
                            width: 100%;
                            height: 100%;
                        }

                    }

                    .mission-name {
                        padding-top: 10px !important;
                        color: cornflowerblue;
                    }

                    .mission-id {
                        padding-bottom: 0 !important;
                    }

                    ol {
                        list-style-type: circle;
                    }

                    .mission-id-res {
                        margin: 0 5px 0 15px;
                        padding-bottom: 10px;
                        color: blueviolet;
                    }
                }
            }

        }
    }

    p span {
        color: blueviolet;
    }

    .spinner-loading {
        margin-top: 70px;
        width: 100%;
        position: absolute;
        z-index: 1;
        top: 50%;
        left: 0;
        bottom: 0;
    }

    .loader {
        border: 16px solid #ea0f0f;
        border-radius: 50%;
        border-top: 16px solid #2223c3;
        width: 100px;
        height: 100px;
        -webkit-animation: spin 2s linear infinite;
        /* Safari */
        animation: spin 2s linear infinite;
        margin: 0 auto;
    }

    /* Safari */
    @-webkit-keyframes spin {
        0% {
            -webkit-transform: rotate(0deg);
        }

        100% {
            -webkit-transform: rotate(360deg);
        }
    }

    @keyframes spin {
        0% {
            transform: rotate(0deg);
        }

        100% {
            transform: rotate(360deg);
        }

    }
}