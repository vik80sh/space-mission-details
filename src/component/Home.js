import React, { Component } from 'react'
import CustomButton from './../atoms/CustomButton';
import { APIs } from './../constant';
import axios from 'axios'
import './Home.scss'

export default class Home extends Component {
    constructor(props) {
        super(props)
        this.state = {
            selectedYear: "",
            isSuccessfulLaunch: "",
            isSuccessfullLand: "",
            flightData: [],
            loader: true
        }
        this.missionYearUI = this.missionYearUI.bind(this);
        this.successfulLaunchUI = this.successfulLaunchUI.bind(this);
        this.successfulLaunchUI = this.successfulLaunchUI.bind(this);
        this.filterFunction = this.filterFunction.bind(this);
        this.cardUI = this.cardUI.bind(this);

    }
    componentDidMount() {
        let data = axios.get(APIs.all);
        data.then(res => {
            this.setState({ flightData: res.data, api: APIs.all, loader: false })
        })
    }
    filterFunction(year, successfullLaunching, successfullLanding) {
        let url = APIs.all;
        let yearFilter = "";
        let landingFilter = "";
        let launchingFilter = "";
        this.setState({ loader: true });
        if (year) {
            yearFilter = '&launch_year=' + year;
            if (this.state.isSuccessfulLaunch)
                launchingFilter = '&launch_success=' + this.state.isSuccessfulLaunch
            if (this.state.isSuccessfullLand)
                landingFilter = '&land_success=' + this.state.isSuccessfullLand
            if (this.state.selectedYear === year) {
                year = "";
                yearFilter = "";
            }
            url = url + launchingFilter + landingFilter + yearFilter;
            let data = axios.get(url);
            data.then(res => {
                this.setState({ flightData: res.data, loader: false, selectedYear: year })
            })
        }
        else if (successfullLaunching) {
            launchingFilter = '&launch_success=' + successfullLaunching;
            if (this.state.selectedYear)
                yearFilter = '&launch_year=' + this.state.selectedYear;
            if (this.state.isSuccessfullLand)
                landingFilter = '&land_success=' + this.state.isSuccessfullLand;
            if (this.state.isSuccessfulLaunch === successfullLaunching) {
                launchingFilter = "";
                successfullLaunching = "";
            }
            url = url + launchingFilter + landingFilter + yearFilter;
            let data = axios.get(url);
            data.then(res => {
                this.setState({ flightData: res.data, loader: false, isSuccessfulLaunch: successfullLaunching })
            })
        } else {
            landingFilter = '&land_success=' + successfullLanding;
            if (this.state.selectedYear)
                yearFilter = '&launch_year=' + this.state.selectedYear;
            if (this.state.isSuccessfulLaunch)
                launchingFilter = '&launch_success=' + this.state.isSuccessfulLaunch
            if (this.state.isSuccessfullLand === successfullLanding) {
                landingFilter = "";
                successfullLanding = "";
            }
            url = url + launchingFilter + landingFilter + yearFilter;
            let data = axios.get(url);
            data.then(res => {
                this.setState({ flightData: res.data, loader: false, isSuccessfullLand: successfullLanding })
            })
        }
    }
    missionYearUI() {
        return <div className="row">
            <p className="filter-title text-center">Launch Year</p>
            {[...Array(15)].map((e, i) => {
                i = 2006 + i;
                let makeButtonBold = i === this.state.selectedYear ? "button-selected" : "";
                return <div className="col-6 text-center" key={i}>
                    <CustomButton text={i} classValue={makeButtonBold} onClick={() => this.filterFunction(i, false, false)} />
                </div>
            })
            }
        </div>
    }
    successfulLaunchUI() {
        return <div className="row successfull-launch">
            <p className="filter-title text-center">Successfull Launch</p>
            <div className="col-6 text-center"><CustomButton text="true" classValue={this.state.isSuccessfulLaunch === "true" ? "button-selected" : ""} onClick={() => this.filterFunction(false, "true", false)} /></div>
            <div className="col-6 text-center"><CustomButton text="false" classValue={this.state.isSuccessfulLaunch === "false" ? "button-selected" : ""} onClick={() => this.filterFunction(false, "false", false)} /></div>
        </div>
    }
    successfullLandUI() {
        return <div className="row successfull-launch">
            <p className="filter-title text-center">Successfull Land</p>
            <div className="col-6 text-center"><CustomButton text="true" classValue={this.state.isSuccessfullLand === "true" ? "button-selected" : ""} onClick={() => this.filterFunction(false, false, "true")} /></div>
            <div className="col-6 text-center"><CustomButton text="false" classValue={this.state.isSuccessfullLand === "false" ? "button-selected" : ""} onClick={() => this.filterFunction(false, false, "false")} /></div>
        </div>
    }

    cardUI() {
        return this.state.flightData.map((data, i) => {
            return <div className="align-items-stretch card-data" key={i}>
                <div className="card-conatainer">
                    <div className="image-wrapper">
                        <img src={data.links.mission_patch} alt={"name " + data.mission_name} />
                    </div>
                    <p className="mission-name"><b>{data.mission_name} </b> </p>
                    <p className="mission-id"><strong>Mission Ids </strong> </p>
                    <ol><li className="mission-id-res">{data.mission_id.length ? data.mission_id[0] : "Not Present"}</li></ol>
                    <p className="launch-year"><strong>Launch Year : </strong><span>{data.launch_year}</span></p>
                    <p className="success"><strong>Successfull Launch : </strong><span>{data.launch_success ? "true" : "false"}</span></p>
                    <p className="landing"><strong>Successfull Landing : </strong><span>{data.rocket.first_stage.cores[0].land_success ? "true" : "false"}</span></p>
                </div>
            </div>

        })
    }
    render() {
        return (
            <section className="row home-wrapper">
                {this.state.loader && <div className="spinner-loading"><div className="loader"></div></div>}
                <div className="col-12 col-sm-4 col-md-3 col-lg-2">
                    <div className="filter-section">
                        <h3>Filters</h3>
                        {this.missionYearUI()}
                        {this.successfulLaunchUI()}
                        {this.successfullLandUI()}
                    </div>
                </div>
                <div className="col-12 col-sm-8 col-md-9 col-lg-10  mission-details">
                    {!this.state.loader && <div className="row card-wrapper">
                        {this.state.flightData && this.state.flightData.length > 0 ? this.cardUI() : <h1>No Result found</h1>}
                    </div>}
                </div>

            </section>
        )
    }
}
