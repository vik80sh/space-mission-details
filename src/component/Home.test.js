import React from 'react';
import { shallow} from 'enzyme';

import Home from "./Home"

describe('Home Page Testing testing ', () => {
    it('testing handlecardname cardvalue',()=>{
        const wrapper = shallow(<Home/>);
        expect(wrapper).toMatchSnapshot();
    });
    it('Filter method testing with year',()=>{
        const wrapper = shallow(<Home/>);
        const instance = wrapper.instance();
        
        instance.filterFunction(2006,false,false);
        expect(wrapper.state().selectedYear).toEqual("");
    });
    it('Filter method testing with SuccessfulLaunch false',()=>{
        const wrapper = shallow(<Home/>);
        const instance = wrapper.instance();
        
        instance.filterFunction(false,"false",false);
        expect(wrapper.state().isSuccessfulLaunch).toEqual("");
    });
    it('Filter method testing with SuccessfulLaunch true',()=>{
        const wrapper = shallow(<Home/>);
        const instance = wrapper.instance();
        
        instance.filterFunction(false,"true",false);
        expect(wrapper.state().isSuccessfulLaunch).toEqual("");
    });
    it('Filter method testing with SuccessfullLand false',()=>{
        const wrapper = shallow(<Home/>);
        const instance = wrapper.instance();
        
        instance.filterFunction(false,false,"false");
        expect(wrapper.state().isSuccessfullLand).toEqual("");
    });
    it('Filter method testing with SuccessfullLand true',()=>{
        const wrapper = shallow(<Home/>);
        const instance = wrapper.instance();
        
        instance.filterFunction(false,false,"true");
        expect(wrapper.state().isSuccessfullLand).toEqual("");
    });
    
})