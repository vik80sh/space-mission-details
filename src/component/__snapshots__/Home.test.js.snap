// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Page Testing testing  testing handlecardname cardvalue 1`] = `ShallowWrapper {}`;
